src/filters.rs

```diff
@@ -269,6 +271,38 @@ pub fn split(input: &Value, args: &[Value]) -> FilterResult {
     }
 }
 
+pub fn join(input: &Value, args: &[Value]) -> FilterResult {
+    // Make sure there is only 1 argument to join
+    if args.len() != 1 {
+        return Err(InvalidArgumentCount(format!("expected 1, {} given", args.len())));
+    }
+
+    match *input {
+        Array(ref array) => {
+            // Not a fan of this, but need to unwrap the Strs and also check for errors
+            let mut strings_to_join = Vec::new();
+            for value in array {
+                match *value {
+                    Str(ref s) => {
+                        strings_to_join.push(s);
+                    }
+                    _ => return Err(InvalidType("Array of Strings expected".to_owned())),
```

Comment:
I wonder if we really have to expect an array of strings or just implicitly convert all values with `value.to_string()`